cheroot/workers/monitor.py

```diff
@@ -0,0 +1,155 @@
+"""Monitor for threadpool."""
+
+import threading
+import time
+
+from .dynpool import DynamicPoolResizer
+
+
+class BackgroundTask(threading.Thread):
+    """A subclass of threading.Thread whose run() method repeats.
+
+    Use this class for most repeating tasks. It uses time.sleep() to wait
+    for each interval, which isn't very responsive; that is, even if you call
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+    """
+
+    def __init__(
+        self,
+        interval,
+        function,
+        args=None,
+        kwargs=None,
+        logger=None,
+    ):
+        """Initialize a BackgroundTask."""
+        super(BackgroundTask, self).__init__()
+        self.interval = interval
+        self.function = function
+        self.args = args if args else []
+        self.kwargs = kwargs if kwargs else {}
+        self.running = False
+        self.log = logger or (lambda msg: None)
+        # default to daemonic
+        self.daemon = True
+
+    def cancel(self):
+        """Cancel the task."""
+        self.running = False
+
+    def run(self):
+        """Run - Invoke function every interval seconds."""
+        self.running = True
+        while self.running:
+            time.sleep(self.interval)
+            if not self.running:
+                return
+            try:
+                self.function(*self.args, **self.kwargs)
+            except Exception as e:
+                self.log(
+                    'Error in background task thread function {}:{}.'.format(
+                        self.function,
+                        e,
+                    ), )
```

Comment:
The point of having a trailing comma is that it's at the very end of the string. It looks weird otherwise.
```suggestion
                    ),
                )
```